Storage/Rust/RustAutofill.swift

```diff
@@ -202,12 +280,102 @@ public class RustAutofill {
         }
     }
 
+    /// Adds an address to the database.
+    ///
+    /// - Parameters:
+    ///   - address: UpdatableAddressFields representing the address to be added.
+    ///   - completion: A closure called upon completion with the added address or an error.
+    public func addAddress(address: UpdatableAddressFields, completion: @escaping (Address?, Error?) -> Void) {
```

Comment:
curious about the completion param types. It seems that we want to either returning an Address or an Error, but never a case of `completion(nil, nil)` or `completion(address, error)`. However, it seems we are allowing those possibilities, so I'm curious  to understand why did we decide to go through using `(Address?, Error?)` as opposed to using a Result type in which we specify a `.failure(error)` or a `.success(address)` case. 

happy to chat live to better understand :) thanks!

Reply:
@cyndichin Good catch!

Blame it on the legacy code refactor but if @jnrahme wants to refactor it in favour of the using proper convention then I would much likely prefer that :)

Reply:
ah ok great, thanks for the clarity nish!

Reply:
https://mozilla-hub.atlassian.net/browse/FXIOS-7810 Ticket created